### PR TITLE
Add new stream types for dynamic

### PR DESCRIPTION
**Pull request by andreivladbrg**
Closes https://github.com/sablier-labs/docs/issues/110

@PaulRBerg I have not added explanations for the graphs yet, as I don't know what specific use cases you have in mind for them. Could you share your ideas here? After that, I'll include the explanations accordingly.

The shapes look like the images below:

<details>
<summary> images </summary>

image

image

image

</details>

### CODE REVIEW

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
The plots look good to me.

Regarding the description for each newly added section - I am sure you can intuit the use cases. I would like to ask you to take a stab at writing the descriptions yourself.

File: docs/concepts/protocol/02-stream-types.mdx [hunk 1]
```diff
@@ -143,3 +145,114 @@ accumulate, so if the recipient doesn't withdraw them, they will be able to with
 The advantage of using Unlock in Steps instead of a normal vesting contract is that Sablier automates the entire
 process. No more worries about setting up vesting contracts or creating a user interface for your employees to claim
 their tokens.
+
+### Monthly Unlocks
+
+Monthly Unlocks is a variation of the Unlock in Steps model, with the distinction of having monthly intervals. This
+model is particularly suited for structured yet flexible token distribution plans such as distributing a set salary
```
Comment (smol-ninja) [inline docs/concepts/protocol/02-stream-types.mdx:152]: How about we say something simple here (it looks a bit complicated to me):

Monthly Unlocks is a special case of Unlock in Steps where assets are unlocked on the same day every month. This is particularly suited for use cases where users want to distribute tokens every month such as distributing salaries.

File: docs/concepts/protocol/02-stream-types.mdx [hunk 2]
```diff
@@ -143,3 +145,114 @@ accumulate, so if the recipient doesn't withdraw them, they will be able to with
...
+/>
+
+Each month, on a preset day, a specific amount of tokens becomes unlocked and available for withdrawal. Like the Unlock
+in Steps model, unwithdrawn tokens will carry over to the next period, providing flexibility and control to the
+recipient. This model is ideal for employers who wish to set up advanced payment schedules for their employees, offering
+them access to funds on a predictable, monthly basis.
+
+### Timelock
+
+The Timelock model enables the locking of a certain amount of tokens for a specified period. This model is designed to
```
Comment (smol-ninja) [inline docs/concepts/protocol/02-stream-types.mdx:181]:
```suggestion
The Timelock model enables locking tokens for a specified period. This model is designed to
```

File: docs/concepts/protocol/02-stream-types.mdx [hunk 3]
```diff
@@ -143,3 +145,114 @@ accumulate, so if the recipient doesn't withdraw them, they will be able to with
...
+        graphType: "polyline",
+        color: "#f77423",
+      },
+    ],
+  }}
+/>
+
+At the beginning of the stream, a fixed amount of tokens is instantly available to the recipient — this is your "signing
+bonus". Following this, the remaining tokens begin to stream continuously at a consistent rate until the end of the
+contract term — this is your "salary".
```
Comment (smol-ninja) [inline docs/concepts/protocol/02-stream-types.mdx:231]: Should we add this as well?

Another use case is a token distribution to investors where a particular amount gets unlocked immediately followed by a linear vesting plan.

File: docs/concepts/protocol/02-stream-types.mdx [hunk 4]
```diff
@@ -143,3 +145,114 @@ accumulate, so if the recipient doesn't withdraw them, they will be able to with
...
+
+### Unlock Cliff
+
+The Unlock Cliff model introduces an unique streaming pattern that combines an initial unlock, a lock phase and then a
+linear unlock phase.
```
Comment (smol-ninja) [inline docs/concepts/protocol/02-stream-types.mdx:236]: How about we add a use case here as well (like all the previous ones)?

This shape is useful for companies who want to distribute tokens to their investors using a cliff followed by linear vesting but also want to unlock some liquidity at the beginning to be able to allow investors to bootstrap AMM pools.

**Review by smol-ninja — COMMENTED** [review]
The descriptions are very useful and easy to understand. I have added my suggestions below. Also a question for you?

- Should we rename "model" to "shape"?

**Comment by andreivladbrg** [comment]
Thanks for the quick review. I agree with your suggestions and have just pushed the changes you recommended.

> Should we rename "model" to "shape"?

I personally like "model"

**Comment by smol-ninja** [comment]
It's failing because of this:

Screenshot 2024-04-17 at 18 50 27

**Comment by PaulRBerg** [comment]
Turns out the problem wasn't the NFT descriptor casing. It was Bun and a missing dev dependency (`react-loadable`). See this:

https://discord.com/channels/398180168688074762/867060369087922187/1191397897598672986

The build works now in CI.

**Review by PaulRBerg — APPROVED** [review]
good to be merged

**Comment by andreivladbrg** [comment]
Thanks for the CI fix, merging now

**Pull request merged**